src/UsgsAstroLsSensorModel.cpp

```diff
@@ -2498,147 +2454,108 @@ double UsgsAstroLsSensorModel::determinant3x3(double mat[9]) const
-
-   for (auto& scanRate : isd.at("line_scan_rate")) {
-     state["m_intTimeLines"].push_back(scanRate[0]);
-     state["m_intTimeStartTimes"].push_back(scanRate[1]);
-     state["m_intTimes"].push_back(scanRate[2]);
-   }
+  state["m_modelName"] = getSensorModelName(isd, parsingWarnings);
+  state["m_imageIdentifier"] = getImageId(isd, parsingWarnings);
+  state["m_sensorName"] = getSensorName(isd, parsingWarnings);
+  state["m_platformName"] = getPlatformName(isd, parsingWarnings);
+  std::cerr << "Model Name Parsed!" << std::endl;
 
-   state["m_detectorSampleSumming"] = isd.at("detector_sample_summing");
-   state["m_startingSample"] = isd.at("detector_line_summing");
-   state["m_ikCode"] = 0;
-   state["m_focal"] = isd.at("focal_length_model").at("focal_length");
-   state["m_zDirection"] = 1;
-   state["m_opticalDistCoef"] = isd.at("optical_distortion").at("radial").at("coefficients");
-   state["m_iTransS"] = isd.at("focal2pixel_samples");
-   state["m_iTransL"] = isd.at("focal2pixel_lines");
+  state["m_focalLength"] = getFocalLength(isd, parsingWarnings);
+  std::cerr << "Focal Length Parsed!" << std::endl;
 
-   state["m_detectorSampleOrigin"] = isd.at("detector_center").at("sample");
-   state["m_detectorLineOrigin"] = isd.at("detector_center").at("line");
-   state["m_detectorLineOffset"] = 0;
+  state["m_nLines"] = getTotalLines(isd, parsingWarnings);
+  state["m_nSamples"] = getTotalSamples(isd, parsingWarnings);
 
-   state["m_dtEphem"] = isd.at("dt_ephemeris");
-   state["m_t0Ephem"] = isd.at("t0_ephemeris");
-   state["m_dtQuat"] =  isd.at("dt_quaternion");
-   state["m_t0Quat"] =  isd.at("t0_quaternion");
+  state["m_iTransS"] = getFocal2PixelSamples(isd, parsingWarnings);
+  state["m_iTransL"] = getFocal2PixelLines(isd, parsingWarnings);
 
-   state["m_numEphem"] = isd.at("sensor_position").at("positions").size();
-   state["m_numQuaternions"] = isd.at("sensor_orientation").at("quaternions").size();
+  state["m_platformFlag"] = 1;
+  state["m_ikCode"] = 0;
+  state["m_zDirection"] = 1;
 
-   {
-     json jayson = isd.at("sensor_position");
-     json unit = jayson.at("unit");
-     unit = unit.get<std::string>();
-
-     for (auto& location : jayson.at("positions")) {
-       state["m_ephemPts"].push_back(metric_conversion(location[0].get<double>(), unit));
-       state["m_ephemPts"].push_back(metric_conversion(location[1].get<double>(), unit));
-       state["m_ephemPts"].push_back(metric_conversion(location[2].get<double>(), unit));
-     }
+  // Zero computed state values
+  state["m_referencePointXyz"] = std::vector<double>(3, 0.0);
 
-     for (auto& velocity : jayson.at("velocities")) {
-       state["m_ephemRates"].push_back(metric_conversion(velocity[0].get<double>(), unit));
-       state["m_ephemRates"].push_back(metric_conversion(velocity[1].get<double>(), unit));
-       state["m_ephemRates"].push_back(metric_conversion(velocity[2].get<double>(), unit));
-     }
-   }
+  // leave these be for now.

```

Comment:
:+1: On "leave these be for now" but what is happening with these state keywords now / in the future?

Reply:
As in, is the intent to change or remove them from the state, or do they just need to be temporarily initialized?

Reply:
I imagine we want to support these in the ISD and copy them into the state like the others. Not 100% sure why they are being hardcoded or to what extent do these values actually matter.

Reply:
Image flip flag - can be removed. It's some weird old code that was in the model.
GSD - I don't know what this is
flying height - This is a computed value, so it shouldn't be in the ISD
half time - This is a computed value, so it shouldn't be in the ISD